### PR TITLE
Improve logging output

### PR DESCRIPTION
**Pull request by Twey**
## Motivation

There is currently a typo in the `linera-service` tests that [results in an error](https://github.com/linera-io/linera-protocol/pull/2257/files#r1686167112) and presumably fails to show debug logs from some parts of our tests.

Our tracing subscriber currently chooses a JSON output automatically if `stderr` is not connected to a TTY; I believe this is sensible behaviour in general but we should be able to override it, since sometimes we want human-readable output from subprocesses whose output is processed.

## Proposal

Fix the typo.  Add a `plain` log format and activate it explicitly in CI.  Set our CI log level to `DEBUG` so we see `INFO` and `DEBUG` messages that can be helpful for our developers to understand what's gone wrong in a CI run.

## Test Plan

Check CI output.

## Release Plan

None needed.

## Links

- [reviewer checklist](https://github.com/linera-io/linera-protocol/blob/main/CONTRIBUTING.md#reviewer-checklist)

### CODE REVIEW

**Review by afck — DISMISSED** [review]
Looks good to me. I think @ma2bd had concerns about it being too noisy, but if I understand correctly, this is already `debug` by default?

**Comment by ma2bd** [comment]
> Looks good to me. I think @ma2bd had concerns about it being too noisy, but if I understand correctly, this is already `debug` by default?

No, the default is INFO.

**Comment by ma2bd** [comment]
@Twey The logs in the CI of this PR look very suspicious

**Comment by Twey** [comment]
> No, the default is `INFO`.

I had thought that @jvff already set it to `DEBUG` a while ago — but you're right, I misremembered.  In fact the level we set explicitly in CI is `WARN`, so I'm not sure why we see `INFO` logs in CI.

I do however think that `DEBUG` is probably correct for CI, since it's targeted at developers.

> @Twey The logs in the CI of this PR look very suspicious

They do, I've scope-creeped this PR to also include the explicit `plain` format we discussed yesterday.  Let's see if it works to make sure we have only plain-format logs in the CI.

**Comment by ma2bd** [comment]
@Twey Now I remember: the problem with the DEBUG level is that GRPC is very verbose.
```
2024-07-23T17:14:00.035512Z DEBUG hyper::client::connect::http: connecting to 127.0.0.1:1235
2024-07-23T17:14:00.035530Z DEBUG Connection{peer=Client}: h2::codec::framed_write: send frame=Data { stream_id: StreamId(29) }
2024-07-23T17:14:00.035535Z DEBUG Connection{peer=Client}: h2::codec::framed_write: send frame=Data { stream_id: StreamId(97) }
2024-07-23T17:14:00.035543Z DEBUG Connection{peer=Client}: h2::c
This step has been truncated due to its large size. Download the full logs from the  menu once the workflow run has completed.
```

**Comment by Twey** [comment]
Oh I set `RUST_LOG=debug` instead of `RUST_LOG=linera=debug`!  Generally we shouldn't see debug output from Hyper and friends.

File: linera-rpc/src/grpc/server.rs [hunk 1]
```diff
@@ -317,7 +317,7 @@ where
 
         for request in actions.cross_chain_requests {
             let shard_id = self.network.get_shard_id(request.target_chain_id());
-            debug!(
+            tracing::trace!(
```
Comment (ma2bd) [inline linera-rpc/src/grpc/server.rs:320]: `trace!` ?

**Review by ma2bd — APPROVED** [review]
Thanks!

**Pull request merged**